src/curl.c

```diff
@@ -150,6 +151,16 @@ static int check_connection(const char *test_capath)
 
 	switch (curl_ret) {
 	case CURLE_OK:
+		if (curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &response) != CURLE_OK) {
+			return -1;
+		}
+		/* when using http an invalid url will still return CURL_OK, we can consider
+		 * responses < 400 valid responses since they mean the request was received
+		 * and understood */
+		if (response >= 400) {
```

Comment:
We don't require that the server responds for requests to / so we can have a 404 here and it's fine.
This function is only checking if the server is up and if we can communicate with the server using ssl (in the case of https).
If it responds anything we're good to go. So don't worry about errors here.

Reply:
good point. as long as we received a response we are good to go 👌

Reply:
ooh no, wait, I remember now why I added this....
If you add a fake http mirror, lets say "http://fakemirror.org/update", during check_connection it is going to respond with curl_ret = CURLE_OK, and response 404. But the server doesn't even exist, so that is why I think we need to check for valid responses.

Reply:
turns out that the proxy I'm connected to is the one responding instead of "http://fakemirror.org/update", this would be solved if using https, so we can basically ignore this at this point and allow it to continue, the command will fail further down the line anyway

Reply:
I'll remove that response check